### DISCUSSION TITLE
make future dataframe ignores periods keyword

### DISCUSSION THREAD

**Discussion by thanosvlo** [ourownstory/neural_prophet#1157] [Q&A - get help using NeuralProphet] [answered]
Hi just a quick question on what is the reasoning behind this line :
https://github.com/ourownstory/neural_prophet/blob/1b446341866dd9866c215808ae51283f7ebed7a6/neuralprophet/forecaster.py#L2808

Id assume - based on the documentation-  that periods control the number of n_forecasts we want to predict into the future. 
Say if the model is trained to predict 7 days into the future and the periods is set to be 3 then we would want to have a prediction for the upcoming 21 days in a rolling inference fashion. 

are there any plans to amend the function to be in line with the documentation and offer the described functionality ? 

Thank you for maintaining this repo, much appreciated.

**Comment by judussoari** [comment, marked as answer]
Hi @thanosvlo 
When we use autoregression (i.e. `self.max_lags > 0`) and call `m.make_future_dataframe()` to predict into the unknown future, then we want to predict a defined number of steps. The number, in that case, is determined by `n_forecasts`. 
We add the parameter `period` for the case of not using autoregression `self.max_lags = 0`. In this case, we allow the user to forecast `periods` steps into the unknown future. 
`periods` and `n_forecasts` are not multiplied. Hope this helps!
